src/services/helpers.ts

```diff
@@ -9,14 +9,47 @@ export class ValidationError extends Error {
   }
 }
 
+export class TimeoutError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TimeoutError";
+  }
+}
+
+async function processTimeout<T>(
+  ms: number,
+  promise: Promise<T>,
+  errorMessage?: string
+): Promise<T> {
+  return new Promise<T>(async (resolve, reject) => {
+    setTimeout(() => {
+      reject(new TimeoutError(errorMessage || "Request timed out"));
+    }, ms);
+    resolve(await promise);
+  });
+}
+
 export async function fetchWithValidator<T, O, I>(
   validator: Type<T, O, I>,
   requestInfo: RequestInfo,
-  init?: RequestInit
+  init?: RequestInit,
+  timeout: boolean | number = false
```

Comment:
```suggestion
  timeout?: boolean | number = false
```

Reply:
oo seb i got an error "Parameter cannot have question mark and initializer.ts(1015)" when i did that because i think it's initialised as false

Reply:
Ok, whoops